python/kfserving/kfserving/storage.py

```diff
@@ -168,16 +171,13 @@ def _download_blob(uri, out_dir: str):  # pylint: disable=too-many-locals
                      account_name,
                      container_name,
                      prefix)
-        try:
-            block_blob_service = BlockBlobService(account_name=account_name)
-            blobs = block_blob_service.list_blobs(container_name, prefix=prefix)
-        except Exception:  # pylint: disable=broad-except
-            token = Storage._get_azure_storage_token()
-            if token is None:
-                logging.warning("Azure credentials not found, retrying anonymous access")
-            block_blob_service = BlockBlobService(account_name=account_name, token_credential=token)
-            blobs = block_blob_service.list_blobs(container_name, prefix=prefix)
+        token = Storage._get_azure_storage_token()
+        if token is None:
+            logging.warning("Azure credentials not found, retrying anonymous access")
+        blob_service_client = BlobServiceClient(account_url, credential=token)
```

Comment:
@maganaluis if the token is None, this should still work right? Does azure support public storage blobs?

Reply:
oh I found the doc https://docs.microsoft.com/en-us/python/api/overview/azure/storage-blob-readme?view=azure-python#types-of-credentials

Reply:
Correct, however that's unlikely to be used. I did verify that Azure AD works. The current documentation still holds well.